src/name/resolve.js

```diff
@@ -32,106 +24,157 @@ module.exports = (createCommon, options) => {
+        .to.eq(`/ipfs/${path}/remainder/file.txt`)
     })
 
-    it('should recursively resolve to an IPFS hash', function (done) {
-      this.timeout(100 * 1000)
+    it('should resolve a record recursive === true with remainder', async () => {
+      const [{ path }] = await ipfs.add(Buffer.from('should resolve a record recursive = true with remainder'))
+
+      const { id: keyId } = await ipfs.key.gen('key-name-remainder', { type: 'rsa', size: 2048 })
 
-      const value = fixture.cid
+      await ipfs.name.publish(path, { 'allow-offline': true })
+      await ipfs.name.publish(`/ipns/${nodeId}`, { 'allow-offline': true, key: 'key-name-remainder' })
+
+      return expect(await ipfs.name.resolve(`/ipns/${keyId}/remainder/file.txt`, { recursive: true }))
+        .to.eq(`/ipfs/${path}/remainder/file.txt`)
+    })
+
+    it('should not get the entry if its validity time expired', async () => {
       const publishOptions = {
-        resolve: false,
-        lifetime: '24h',
+        lifetime: '100ms',
         ttl: '10s',
-        key: 'self'
+        'allow-offline': true
       }
 
-      // Generate new key
-      ipfs.key.gen(keyName, { type: 'rsa', size: 2048 }, (err, key) => {
-        expect(err).to.not.exist()
+      // we add new data instead of re-using fixture to make sure lifetime handling doesn't break
+      const [{ path }] = await ipfs.add(Buffer.from('should not get the entry if its validity time expired'))
+      await ipfs.name.publish(path, publishOptions)
+      await delay(500)
+      // go only has 1 possible error https://github.com/ipfs/go-ipfs/blob/master/namesys/interface.go#L51
+      // so here we just expect an Error and don't match the error type to expiration
+      try {
+        await ipfs.name.resolve(nodeId)
+      } catch (error) {
+        expect(error).to.exist()
+      }
+    })
+  })
+
+  describe('.name.resolve dns', function () {
+    const common = createCommon()
+    let ipfs
+    this.retries(3)
 
-        keyId = key.id
+    before(function (done) {
+      common.setup((err, factory) => {
+        expect(err).to.not.exist()
 
-        // publish ipfs
-        ipfs.name.publish(value, publishOptions, (err, res) => {
+        spawnNodeWithId(factory, (err, node) => {
           expect(err).to.not.exist()
-          expect(res).to.exist()
 
-          publishOptions.key = keyName
+          ipfs = node
+          done()
+        })
+      })
+    })
 
-          // publish ipns with the generated key
-          ipfs.name.publish(`/ipns/${nodeId}`, publishOptions, (err, res) => {
-            expect(err).to.not.exist()
-            expect(res).to.exist()
+    after((done) => common.teardown(done))
 
-            const resolveOptions = {
-              nocache: false,
-              recursive: true
-            }
+    it('should resolve /ipns/ipfs.io', async () => {
+      return expect(await ipfs.name.resolve('/ipns/ipfs.io'))
+        .to.match(/\/ipfs\/.+$/)
```

Comment:
Can we use `is-ipfs`?

Reply:
we cant for the remainder tests i would prefer to keep them all the same